src/libraries/System.Diagnostics.DiagnosticSource/src/System/Diagnostics/Metrics/CounterAggregator.cs

```diff
@@ -1,13 +1,24 @@
 ﻿// Licensed to the .NET Foundation under one or more agreements.
 // The .NET Foundation licenses this file to you under the MIT license.
 
+using System.Runtime.InteropServices;
+using System.Threading;
+using Internal;
+
 namespace System.Diagnostics.Metrics
 {
     internal sealed class CounterAggregator : Aggregator
     {
         private readonly bool _isMonotonic;
-        private double _delta;
         private double _aggregatedValue;
+        /// <summary>Per-core deltas.</summary>
+        /// <remarks>
+        /// Stored as an array of deltas rather than a single delta to reduce contention from
+        /// highly-parallel updates. The elements are padded to reduce false sharing.
+        /// The array is limited to a semi-arbitrary limit of 8 in order to avoid excessive memory
+        /// consumption when many counters are being used.
+        /// </remarks>
+        private readonly PaddedDouble[] _deltas = new PaddedDouble[Math.Min(Environment.ProcessorCount, 8)];
```

Comment:
@stephentoub shouldn't this be a max if the comment is accurate?

Reply:
I believe they're both correct. We want to increase the number up to the processor count but not go above 8, so we want the processor count or 8, whichever is lower.

Reply:
@bencyoung-Fignum min is ensuring limit, here it means 8 or less.

Reply:
Doh you are right, apologies!